contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -133,7 +135,7 @@ contract SmartAccount is
     // Getters
 
     function domainSeparator() public view returns (bytes32) {
-        return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, getChainId(), this));
+        return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, block.chainid, this));
```

Comment:
In one of my commits I have still kept the read method getChainId and made chainId immutable which only gets set once in the constructor using block.chainid. should we keep it that way post merge?
this is good too, just leaving a note for conflicts and discussion point.

Reply:
I think we of course should keep getChainId() method for external apps
as for setting it just in the constructor, is sload cheaper than block.chainid?